### PR TITLE
Updated Nil-Checking section of the README.

### PR DESCRIPTION
**Pull request by lpollard11**
Updated the Nil-Checking section of the README with three new cases.
Included standard for the below cases:
-If we want to enter a scope if at least one object is non-nil
-If we want to enter a scope if the object is nil
-If we want to enter a scope if the an object is non-nil/nil and a bool is true/false.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -519,6 +519,30 @@ func isError(error: Error?) -> Bool {
...
+```swift
+if item == nil {
+	// Do stuff..
+}
+```
+
+Additionally, check directly for nil if you are checking for nil AND checking if a bool is YES or NO.
+
+```swift
+if item != nil && propertyBool == true {
```
Comment (Haud) [inline README.md]: I'm not sure this case is as clear cut as the other two. I think it'd be better as:

`if let _ = nil where propertyBool`

Comment (lpollard11) [inline README.md]: That's definitely makes sense if you are only checking one propertyBool. How would we use this method if we are checking multiple propertyBools in addition to something that might be nil?

Comment (Haud) [inline README.md]: `if let _ = nil where testOne && testTwo`

Comment (Haud) [inline README.md]: where `testOne`, `testTwo` are `Bool`

Comment (lpollard11) [inline README.md]: I tested that in playground and I'm getting errors. Plus, that would only work if I want to test if both `testOne` and `testTwo` are `true` or `false`. What if I want to check if `testOne` is `false` and `testTwo` is `true`?

Comment (Haud) [inline README.md]: Can you tell me what errors you are getting. This works for me in playground:

let testOne = true
let testTwo = false

if let str = str where testOne && !testTwo {

}

Comment (lpollard11) [inline README.md]: It works! Nice, I'll update my pr now

**Comment by paulmiard** [comment]
Feels more like best practices, or even tips to me. Good tips. Yet not sure it should live there

**Comment by Haud** [comment]
I'm ok with either approach; Adding more examples for various use cases to the ruleset is perfectly valid IMO

**Comment by paulmiard** [comment]
Ok. I'm not against it either. And it makes more sense to have it right after the rule. But there is a difference between rule and examples, and as the guide grows it might get to a point it's too big. We'll see if/when we get there

File: README.md [hunk 2]
```diff
@@ -519,6 +519,30 @@ func isError(error: Error?) -> Bool {
 }
 ```
 
+If there's a case where you want to enter a scope if AT LEAST ONE item is non-nil then use the `??` operator.
+
+```swift
+if let _ = item ?? item1 {
```
Comment (ThibaultKlein) [inline README.md]: is that something Apple is recommending? In Swift 3 for example `--` and `++` are deprecated, just curious

File: README.md [hunk 3]
```diff
@@ -519,6 +519,30 @@ func isError(error: Error?) -> Bool {
...
+```swift
+if let _ = item where propertyBoolOne && !propertyBoolTwo {
+	// Bool properties have been evaluated.
+}
+```
+
+If you want to enter a scope, only if an object is nil, then you can check directly for nil.
+
+```swift
+if item == nil {
```
Comment (ThibaultKlein) [inline README.md]: is that necessary? I think here you are just explaining how to use a feature, not defining a best practice, thoughts?

File: README.md (hunk 2 above)
Comment (Haud) [inline README.md]: null-coalescing operator is not going away,

Comment (ThibaultKlein) [inline README.md]: IMHO this shouldn't be a standard, it's a good operator to save some lines but it doesn't provide more clarity to the code (it's even the opposite if you are not familiar with it). We shouldn't force developers to do it.

**Comment by ThibaultKlein** [comment]
@lpollard11 would you mind moving this to the `Best Practices` section and create a link from the style guide nil-checking rule?

**Comment by lpollard11** [comment]
Yeah for sure! @ThibaultKlein Do you want me to move all of it or just part of it?

**Comment by ThibaultKlein** [comment]
Move it all please :)

**Comment by Haud** [comment]
@lpollard11 @ThibaultKlein I'd also maybe add a link in this section referring to it in the best practices so  we can point it out. I think it's valuable information and my worry is that if we don't directly link back to the best practices guide, no one will ever use it / know of it.

File: README.md (hunk 3 above)
Comment (lpollard11) [inline README.md]: Well I wanted to include a case where we don't want to use `if let` to check for nil and its fine to directly check for nil

File: README.md (hunk 2 above)
Comment (lpollard11) [inline README.md]: I wanted a way to enter a scope if at least one object in a series of objects is not nil using `if let`. In that case should we support just checking directly for nil?

**Comment by paulmiard** [comment]
That's something we might do for other rules as well, would be good to have a format for it (the same way we have the _rationale_ thing. Give it a try we'll see how it looks

**Comment by lpollard11** [comment]
Sounds good all, I'll move it all to the best practice and include a link to that in this nil-checking section in the actual style guide.

File: README.md (hunk 2 above)
Comment (ThibaultKlein) [inline README.md]: I'm good having that in the best practices

File: README.md (hunk 3 above)
Comment (ThibaultKlein) [inline README.md]: same comment, i'm good if it's part of the best practices

File: README.md [hunk 4]
```diff
@@ -519,6 +519,8 @@ func isError(error: Error?) -> Bool {
 }
 ```
 
+For style suggestions regarding nil checking visit our [best practices](https://github.com/prolificinteractive/swift-style-guide/blob/master/BestPractices.md) section.
```
Comment (ThibaultKlein) [inline README.md:522]: `[Best Practices]` I'd assume for better styling

**Comment by ThibaultKlein** [comment]
I left a comment, still approving since it's not a huge deal

:+1:

**Comment by Haud** [comment]
:+1:

**Pull request merged**